zio-http/src/main/scala/zhttp/http/Middleware.scala

```diff
@@ -113,6 +113,38 @@ object Middleware {
   def basicAuth(u: String, p: String): Middleware[Any, Nothing] =
     basicAuth { case (user, password) => (user == u) && (password == p) }
 
+  def addCookieM[R, E](cookie: ZIO[R, E, Cookie]): Middleware[R, E] =
+    patchM((_, _) => cookie.mapBoth(Option(_), c => Patch.addHeader(Headers.setCookie(c))))
+
+  /**
+   * CSRF middlewares : To prevent Cross-site request forgery attacks. This middleware is modeled after the double
+   * submit cookie pattern.
+   * @method
+   *   csrfGenerate - Sets cookie with CSRF token
+   * @method
+   *   csrfValidate - Validate token value in request headers against value in cookies
+   * @see
+   *   https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#double-submit-cookie
+   */
+
+  def csrfGenerate[R, E](
+    tokenGen: ZIO[R, Nothing, String] = random.nextUUID.map(_.toString),
```

Comment:
```suggestion
    tokenGen: ZIO[R, Nothing, String] = UIO(java.util.UUID.randomUUID.toString),
```

Reply:
We won't be able to test this if we make this change.